### ISSUE TITLE
Temporary fix for failing to sync files

### ISSUE THREAD

**Issue by albinahlback** [cross-platform-actions/action#51]
I have a problem where, occasionally, the runner cannot sync the files before exiting the emulator. The job seems to work fine, it is just this issue.

As a temporary fix (as I know that this is already reported as an issue in #29), I would like to indicate to the runner that some steps was completed, and that a timeout and error is fine. I did this by setting
```yml
  my-freebsd-job:
    name: FreeBSD
    runs-on: macos-latest
    steps:
      - uses: cross-platform-actions/action@v0.10.0
        timeout-minutes: 25
        continue-on-error: true
        # Create some files if successful
      - name: "next step"
        # Check if the files exist
```
However, if it did time out, the runner will not find the files, regardless of whether it was successful or not. I have also tried to set output variables via `echo "myvar=true" >> $GITHUB_OUTPUT`, but that should also save the variables to a file, so it should also not work.

Perhaps `cross-platform-actions/action` works inside another directory? Perhaps these files exists, just that they are in another directory? If so, what is its path?

**Comment by jacob-carlborg** [comment]
Hmm, I looked at https://github.com/flintlib/flint2/actions/runs/4497220469/jobs/7912619166?pr=1307. It seems like it timed out while syncing the files from the VM to the runner. I'm guessing the files were never synced.

> Perhaps cross-platform-actions/action works inside another directory? Perhaps these files exists, just that they are in another directory? If so, what is its path?

No, it uses the default working directory for the runner.

**Comment by albinahlback** [comment]
Yeah, happens quite often. So if the job is aborted, will it still remove files? Or where do the files go to? Anyway, feel free to close.

**Comment by jacob-carlborg** [comment]
> So if the job is aborted, will it still remove files? Or where do the files go to?

There is no live syncing of files. Files are synced once in the beginning from the runner to the VM when the VM is ready. Then, when the command is complete, files are synced once from the VM to the runner. If the last sync is not completed before the job terminates, then there might be missing files.

**Comment by kobalicek** [comment]
BTW I'm playing with `sync_files: runner-to-vm` and this didn't fix any issues for me. I still have shutdown failures unfortunately.

**Comment by jacob-carlborg** [comment]
Closing as no longer applicable.

This was filed as a workaround for #29, which has now been closed: the underlying sync/timeout failures were specific to macOS runners and the Xhyve hypervisor, neither of which is supported anymore. Linux is the only supported host platform and QEMU the only hypervisor.

To answer the original question for the record: the action uses the runner's default working directory, so there is no other location where the files end up. Syncing happens twice — once from runner to VM when the VM is ready, and once from VM to runner after the command completes — so if the job is killed before that second sync, the files simply never make it back.

If file syncing still fails on a current version, please open a new issue with a link to the failing job.

---
_Generated by [Claude Code](https://claude.ai/code)_
